### DISCUSSION TITLE
how do get the current node on click?

### DISCUSSION THREAD

**Discussion by tyler-g** [ueberdosis/tiptap#2826] [General Questions]
This seems like a pretty common use-case, but I cannot seem to find a simple answer on how to do this.

Basically, on click (not on select), so essentially using the `onFocus` `event` I would like to access the current node that you've clicked on.  I need access to the `marks` on that node to be able to check for something.  What is the correct way to do this?  Much thanks

**Comment by sereneinserenade** [comment]
you can do something like `selection.$head.parent` which will give you the current active node. https://prosemirror.net/docs/ref/#state.Selection.$head -> https://prosemirror.net/docs/ref/#model.ResolvedPos.parent

**Comment by Jayraj-R** [comment]
I want to get current node type(Paragraph, Heading, table celll...) on the basis of cursor position. Is that possible ?

**Reply by alvintheodora** [reply]
I used this to select the node selection  `props.editor.commands.setNodeSelection(props.getPos())`
